lib/common/xml.c

```diff
@@ -845,6 +845,61 @@ pcmk__xml_is_name_start_char(const char *utf8, int *len)
            || ((c >= 0x10000) && (c <= 0xEFFFF));
 }
 
+/*!
+ * \internal
+ * \brief Check whether the first character of a string is an XML NameChar
+ *
+ * See https://www.w3.org/TR/xml/#NT-NameChar.
+ *
+ * This is almost identical to libxml2's \c xmlIsDocNameChar(), but they don't
+ * expose it as part of the public API.
+ *
+ * \param[in]  utf8  UTF-8 encoded string
+ * \param[out] len   Where to store size in bytes of first character in \p utf8
+ *
+ * \return \c true if \p utf8 begins with a valid XML NameChar, or \c false
+ *         otherwise
+ */
+bool
+pcmk__xml_is_name_char(const char *utf8, int *len)
+{
+    int c = 0;
+
+    // See comment regarding len in pcmk__xml_is_name_start_char()
+    *len = 4;
+
+    // Note: xmlGetUTF8Char() assumes a 32-bit int
+    c = xmlGetUTF8Char((pcmkXmlStr) utf8, len);
+
+    if (c < 0) {
+        crm_err("Invalid UTF-8 character 0x%X", c);
```

Comment:
info or debug

Reply:
I figured error because in practice (outside of unit testing) we should never hit this. It implies a UTF-8 encoding error.

With that said, printing `c` is unhelpful. In this case `c == -1`, not the invalid sequence of bytes.